File: noop-java/src/main/java/com/exyui/android/debugbottle/components/injectors/IntentInjector.java
package com.exyui.android.debugbottle.components.injectors;

import android.content.Intent;
import android.util.Log;

/**
 * Created by yuriel on 9/7/16.
 */

@SuppressWarnings("unused")
public class IntentInjector {

    public final static IntentInjector INSTANCE = new IntentInjector();
    private static final String TAG = "FakeIntentInjector";
    private IntentInjector() {}

    public void setActivity(Object a) {
        Log.d(TAG, "setActivity");
    }

    public void put(Object a, Object b) {
        Log.d(TAG, "put");
    }

    public Intent get() {
        return null;
    }

    public void run(Object a)  {
        Log.d(TAG, "run");
    }
}

File: noop/src/main/kotlin/me/chunyu/yuriel/kotdebugtool/components/injector/IntentInjector.kt
package me.chunyu.yuriel.kotdebugtool.components

import android.content.Intent
import android.util.Log

/**
 * Created by yuriel on 8/30/16.
 */
object IntentInjector {
    private val TAG = "FakeIntentInjector"

    fun setActivity(any: Any) {
        Log.d(TAG, "setActivity")
    }

    fun add(vararg any: Any) {
        Log.d(TAG, "add")
    }

    fun get(any: Any): Intent? = null

    fun run(any: Any) {
        Log.d(TAG, "run")
    }

}